test/adt_test.exs

```diff
@@ -61,4 +61,12 @@ defmodule AdtTest do
       _ -> false
     end)
   end
+
+  test "adtcase allows per-module pattern matching" do
+    foo = %AdtDefinition.Foo{}
+    assert 0 == (ADT.case AdtDefinition, foo do
```

Comment:
That seems okay to me. Also means there's no need to check if the module exists and is an ADT inside of the macro.